app/javascript/react/components/Calendar/Month.tsx

```diff
@@ -0,0 +1,85 @@
+import React from "react";
+import moment from "moment";
+
+import * as S from "./Month.style";
+
+import { yellow } from "../../assets/styles/colors";
+
+const getFullWeeksOfMonth = (year: number, month: number) => {
```

Comment:
I am not against it, but looking at the design I can see we should always end the month on the last day of the considered month, do not go further, but now we do. Please consider the following change, it is not styled and improved but shows the concept

```
const getFullWeeksOfMonth = (year: number, month: number) => {
  let currentMonth = moment([year, month]).startOf("month");
  const endOfCurrentMonth = moment(currentMonth).endOf('month');

  let daysDividedBySeven = currentMonth.daysInMonth()
  while (daysDividedBySeven % 7 != 0) { daysDividedBySeven += 1 }

  let daysDifference = daysDividedBySeven - currentMonth.daysInMonth()
  let firstConsideredDay = currentMonth.subtract(daysDifference, "days")

  let weeks = [];
  let currentDay = firstConsideredDay;

  while (currentDay.isBefore(endOfCurrentMonth)) {
    let week = [];

    for (let i = 0; i < 7; i++) {
      let x = firstConsideredDay.clone().add(i, "days").format("YYYY-MM-DD");
      week.push(x);
    }

    weeks.push(week);
    firstConsideredDay.add(1, "week");
  }

  return weeks;
};
```